packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditForm.tsx

```diff
@@ -23,23 +26,23 @@ export const StocktakeLineEditForm: FC<StocktakeLineEditProps> = ({
 }) => {
   const t = useTranslation(['common', 'inventory']);
   const { items } = useStocktakeRows();
+  const disabled = mode === ModalMode.Update;
 
   return (
     <>
       <ModalRow>
         <ModalLabel label={t('label.item')} />
         <Grid item flex={1}>
-          <ItemSearchInput
+          <StockItemSearchInput
             autoFocus={!item}
-            disabled={mode === ModalMode.Update}
+            disabled={disabled}
             currentItemId={item?.id}
             onChange={onChangeItem}
-            extraFilter={itemToCheck => {
-              const itemAlreadyInStocktake = items?.some(
-                ({ item }) => itemToCheck.id === item?.id
-              );
-              return !itemAlreadyInStocktake;
-            }}
+            extraFilter={
+              disabled
+                ? undefined
+                : item => !items?.some(({ id }) => id === item.id)
+            }
```

Comment:
Changed to `StockItemSearchInput` and offroad fixed the warning about the value being invalid (when editing a line, it was filtering out the item but setting it as the value).

I did this for all request, stocktake, outbound, inbound but not response as that item is non-editable